R/zzz.R

```diff
@@ -16,6 +16,35 @@ R7_object <- new_class(
 )
 methods::setOldClass("R7_object")
 
+#' @export
+`$.R7_object` <- function(x, name) {
+  if (typeof(x) %in% c("list", "environment")) {
+    NextMethod()
+  } else {
+    msg <- sprintf(
+      "Can't access R7 properties with $. Did you mean %s@%s?",
+      deparse1(substitute(x)),
+      name
+    )
+    stop(msg, call. = FALSE)
+  }
+}
+#' @export
+`$<-.R7_object` <- function(x, name, value) {
+  if (typeof(x) %in% c("list", "environment")) {
+    NextMethod()
+  } else {
+    msg <- sprintf(
+      "Can't access R7 properties with $. Did you mean ...@%s <- %s?",
```

Comment:
```suggestion
      "Can't set R7 properties with `$<-`. Did you mean `...@%s <- %s`?",
```

^ Not sure how precise you want to be here

Does `deparse1(substitute(x))` give `*tmp*` or something like that here? I was somehow expecting that to be used too

Reply:
Yeah, I tried it and (somewhat obviously) it does give `*tmp*` so it's not useful